src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -57,14 +57,33 @@ internal class EventBasedLRUCache<TKey, TValue> : IDisposable
         private readonly double _compactionPercentage = .20;
         private LinkedList<LRUCacheItem<TKey, TValue>> _doubleLinkedList = new LinkedList<LRUCacheItem<TKey, TValue>>();
         private BlockingCollection<Action> _eventQueue = new BlockingCollection<Action>();
-        private readonly Task _eventQueueTask;
+
+        // the event queue and maintenance tasks
+        private Task _eventQueueTask;
+
         private ConcurrentDictionary<TKey, LRUCacheItem<TKey, TValue>> _map;
         // When the current cache size gets to this percentage of _capacity, _compactionPercentage% of the cache will be removed.
         private readonly double _maxCapacityPercentage = .95;
         private bool _disposed = false;
         private readonly int _tryTakeTimeout;
         // if true, expired values will not be added to the cache and clean-up of expired values will occur on a 5 minute interval
         private readonly bool _removeExpiredValues;
+        private readonly int _cleanUpIntervalInMilliSeconds;
+
+        private readonly TaskCreationOptions _options;
+
+        // task states used to ensure thread safety (Interlocked.CompareExchange)
+        private const int EventQueueTaskStopped = 0; // task not started yet
+        private const int EventQueueTaskRunning = 1; // task is running
+        private const int EventQueueTaskStopRequested = 2; // a request has been received to stop the task
+
+        private int _eventQueueTaskState = EventQueueTaskStopped;
+
+        // timer that removes expired items periodically
+        private Timer _timer = null;
+
+        //////// for testing purpose only to verify the task count
```

Comment:
Will this be kept when the PR is merged in? Wondering why it's commented out this way.